File: src/two-pointers/python3/28_FindTheIndexOfTheFirstOccurrenceInAString.py
#
# @lc app=leetcode id=28 lang=python3
#
# [28] Find the Index of the First Occurrence in a String
#

# @lc code=start
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        result=-1
        if len(haystack) >= len(needle):
            for i in range(0,len(haystack)-len(needle)+1,1):
                if haystack[i] == needle[0]:
                    needle_index=1
                    while needle_index < len(needle):
                        if haystack[i+needle_index] == needle[needle_index]:
                            needle_index=needle_index+1
                        else:
                            break
                    if needle_index == len(needle):
                        result=i
                        break
                        
        return result
        
# @lc code=end

File: work/typescript/string/28_FindTheIndexOfTheFirstOccurrenceInAString.ts
/*
 * @lc app=leetcode id=28 lang=typescript
 *
 * [28] Find the Index of the First Occurrence in a String
 */

// @lc code=start
function strStr(haystack: string, needle: string): number {
    const len_needle:number=needle.length
    let i:number=0;
    while(i <= haystack.length-len_needle){
        if(haystack.slice(i,i+len_needle)==needle){
            return i;
        }
        else{
            i++;
        };
    };
    return -1;
};
// @lc code=end

